common/views/components/Header/HeaderSearch.tsx

```diff
@@ -0,0 +1,96 @@
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { useRouter } from 'next/router';
+import { classNames } from '@weco/common/utils/classnames';
+import SearchBar from '@weco/common/views/components/SearchBar/SearchBar';
+import Space from '@weco/common/views/components/styled/Space';
+import { getQueryPropertyValue } from '@weco/common/utils/search';
+
+const Overlay = styled.div.attrs<{ isActive: boolean }>(props => ({
+  className: classNames({
+    'is-hidden': !props.isActive,
+  }),
+}))<{ isActive: boolean }>`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.6);
+  z-index: 9999;
+`;
+
+const SearchBarWrapper = styled(Space).attrs({
+  v: { size: 'xl', properties: ['padding-bottom'] },
+})`
+  position: absolute;
+  width: 100%;
+  ${props => props.theme.makeSpacePropertyValues('l', ['padding-top'])};
+  background-color: ${props => props.theme.color('white')};
+`;
+
+type Props = {
+  isActive: boolean;
+  setIsActive: Dispatch<SetStateAction<boolean>>;
+};
+
+const HeaderSearch = ({ isActive, setIsActive }: Props) => {
+  const router = useRouter();
+  const routerQuery = getQueryPropertyValue(router?.query?.query);
+  const [inputValue, setInputValue] = useState(routerQuery || '');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsActive(false);
+    setInputValue('');
+  }, [router?.pathname, router?.query]);
+
+  useEffect(() => {
+    if (isActive) {
+      inputRef?.current?.focus();
+    }
+  }, [isActive]);
+
+  useEffect(() => {
+    function handleEscapeKey(event: KeyboardEvent) {
+      if (event.code === 'Escape') {
+        setIsActive(false);
+      }
+    }
+
+    document.addEventListener('keydown', handleEscapeKey);
+    return () => document.removeEventListener('keydown', handleEscapeKey);
+  }, []);
+
+  return (
+    <Overlay
+      isActive={isActive}
+      onClick={() => {
+        setIsActive(false);
+      }}
+    >
+      <SearchBarWrapper onClick={e => e.stopPropagation()}>
+        <form
+          className="container"
+          id="global-search-form"
+          onSubmit={event => {
```

Comment:
I might be missing something, but can we just use the `GET` [method](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-method) of the form (and name the input appropriately) to do this without any JS at all?

Reply:
Nice one 🎉 thank you